src/scripts/treelite_python/score.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+LightGBM/Python inferencing script
+"""
+import os
+import sys
+import argparse
+import lightgbm
+import numpy
+from distutils.util import strtobool
+
+# let's add the right PYTHONPATH for common module
+COMMON_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))
+
+if COMMON_ROOT not in sys.path:
+    print(f"Adding {COMMON_ROOT} to PYTHONPATH")
+    sys.path.append(str(COMMON_ROOT))
+
+# before doing local import
+from common.metrics import MetricsLogger
+from common.io import input_file_path
+
+
+def get_arg_parser(parser=None):
+    """Adds component/module arguments to a given argument parser.
+
+    Args:
+        parser (argparse.ArgumentParser): an argument parser instance
+
+    Returns:
+        ArgumentParser: the argument parser instance
+
+    Notes:
+        if parser is None, creates a new parser instance
+    """
+    # add arguments that are specific to the module
+    if parser is None:
+        parser = argparse.ArgumentParser(__doc__)
+
+    group_i = parser.add_argument_group("Input Data")
+    group_i.add_argument("--data",
+        required=True, type=input_file_path, help="Inferencing data location (file path)")
+    group_i.add_argument("--nthreads",
+        required=False, default = 1, type=int, help="number of threads")
+    group_i.add_argument("--model",
+        required=False, type=input_file_path, help="Exported model location (file path)")
+    group_i.add_argument("--output",
+        required=False, default=None, type=str, help="Inferencing output location (file path)")
+    
+    group_params = parser.add_argument_group("Scoring parameters")
+    group_params.add_argument("--predict_disable_shape_check",
+        required=False, default=False, type=strtobool, help="See LightGBM documentation")
+    
+    return parser
+
+
+def run(args, other_args=[]):
+    """Run script with arguments (the core of the component)
+
+    Args:
+        args (argparse.namespace): command line arguments provided to script
+        unknown_args (list[str]): list of arguments not known
+    """
+    # create sub dir and output file
+    if args.output:
+        os.makedirs(args.output, exist_ok=True)
+        args.output = os.path.join(args.output, "predictions.txt")
+
+    # initializes reporting of metrics
+    metrics_logger = MetricsLogger("treelite.score")
+
+    # add some properties to the session
+    metrics_logger.set_properties(
+        framework = 'treelite_python',
+        task = 'score',
+        lightgbm_version = lightgbm.__version__
+    )
+
+    import pandas as pd
```

Comment:
Could you add those in the header section of the code?

This way, we can catch missing dependencies during unit tests